build/cmake/CMakeLists.txt

```diff
@@ -11,15 +11,14 @@ cmake_minimum_required(VERSION 2.8.9 FATAL_ERROR)
   
 # As of 2018-12-26 ZSTD has been validated to build with cmake version 3.13.2 new policies. 
 # Set and use the newest cmake policies that are validated to work 
-set(ZSTD_MAX_VALIDATED_CMAKE_VERSION "3.13.2") 
-if("${CMAKE_MAJOR_VERSION}" LESS 3) # Cmake version <3.7 does not understand the VERSION_LESS_EQUAL operator
+set(ZSTD_MAX_VALIDATED_CMAKE_MAJOR_VERSION "3") 
+set(ZSTD_MAX_VALIDATED_CMAKE_MINOR_VERSION "13") #Policies never changed at PATCH level
+if("${CMAKE_MAJOR_VERSION}" LESS 3)
   set(ZSTD_CMAKE_POLICY_VERSION "${CMAKE_VERSION}") 
-elseif(("${CMAKE_MAJOR_VERSION}" EQUAL 3 AND "${CMAKE_MINOR_VERSION}" GREATER 6) OR ("${CMAKE_MAJOR_VERSION}" GREATER 3))
-  if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "${ZSTD_MAX_VALIDATED_CMAKE_VERSION}") 
+elseif( "${ZSTD_MAX_VALIDATED_CMAKE_MINOR_VERSION}" GREATER "${CMAKE_MINOR_VERSION}")
```

Comment:
What if, sometimes in the future, some `4.0` version gets released ?
Would it use this branch ?

Reply:
No, not until zstd is verified to work with 4.0+, and then we would need to update lines 14 & 15 to the latest validated version.

In fact if new policies are added in cmake 3.14, those would not be enforced until they are validated and lines 14 & 15 are updated.

Reply:
I'm concerned by the following scenario : 
if a `4.0` version exist one day, I expect : 
- `CMAKE_MAJOR_VERSION == 4`
- `CMAKE_MINOR_VERSION == 0`

Consequently, first test `("${CMAKE_MAJOR_VERSION}" LESS 3)` is false,
but second test `( "${ZSTD_MAX_VALIDATED_CMAKE_MINOR_VERSION}" GREATER "${CMAKE_MINOR_VERSION}")` is true
because `ZSTD_MAX_VALIDATED_CMAKE_MINOR_VERSION =13` which is `>` to `CMAKE_MINOR_VERSION == 0`.

Hence the second branch is taken,
which results in `set(ZSTD_CMAKE_POLICY_VERSION "${CMAKE_VERSION}") ` with `CMAKE_VERSION == 4.0`,
and I suspect that's not an intended outcome, since you want policy pinned to `3.13`.